### PR TITLE
Remove _type term filters from cluster alert watches (#38819)

### PR DESCRIPTION
**Pull request by ycombinator**
Backport of https://github.com/elastic/elasticsearch/pull/38819. Original message:

This PR removes usages of the `_type` field in `_search` requests issued from Monitoring code.

### CODE REVIEW

**Comment by ycombinator** [comment]
@jpountz @bleskes This PR is a backport to the `7.0` branch so @jakelandis suggested to get your approval before merging it, since it's past 7.0 FF.

**Comment by jakelandis** [comment]
> @jpountz @bleskes This PR is a backport to the `7.0` branch so @jakelandis suggested to get your approval before merging it, since it's past 7.0 FF.

Just wanted raise awareness that we are proceeding with the removal of types for monitoring, targeting (at least) 7.0 GA. (likely rc1)  This happens to be the first of a few related PR's, please thumbs up to approve this initiative back 7.0 (no need to review code, we are doing that on the original PRs).

**Review by jakelandis — APPROVED** [review]

**Pull request merged**
